tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,838 @@
+from ocs_ci.ocs.exceptions import CommandFailed, ResourceWrongStatusException
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+import tempfile
+import re
+
+from ocs_ci.ocs.resources.pod import (
+    get_mon_pods,
+    get_osd_pods,
+    get_deployments_having_label,
+    get_mds_pods,
+)
+from ocs_ci.ocs.resources import pod
+
+from ocs_ci.ocs import ocp, constants, defaults
+from ocs_ci.ocs.ocp import OCP
+
+from ocs_ci.helpers.helpers import wait_for_resource_state
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import run_cmd, exec_cmd
+
+logger = logging.getLogger(__name__)
+
+
+@tier3
+@ignore_leftovers
+@pytest.mark.polarion_id("")
+@pytest.mark.bugzilla("1973256")
+@skipif_ocs_version("<4.9")
+@skipif_openshift_dedicated
+@skipif_external_mode
+class TestMonitorRecovery(E2ETest):
+    """
+    Test to verify monitor recovery
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        project_factory,
+        multi_pvc_factory,
+        dc_pod_factory,
+        mcg_obj,
+        bucket_factory,
+    ):
+        """
+        Creates pvcs, pods and obcs
+
+        """
+        self.filename = "sample_file.txt"
+        self.sanity_helpers = Sanity()
+        project = project_factory()
+        rbd_pvcs = multi_pvc_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            project=project,
+            size=1,
+            access_modes=[constants.ACCESS_MODE_RWO],
+            num_of_pvc=1,
+            wait_each=True,
+        )
+
+        cephfs_pvcs = multi_pvc_factory(
+            interface=constants.CEPHFILESYSTEM,
+            project=project,
+            size=1,
+            access_modes=[constants.ACCESS_MODE_RWX],
+            num_of_pvc=1,
+        )
+
+        # Create deployment-config based pods
+        self.dc_pods = []
+        for rbd_pvc in rbd_pvcs:
+            self.dc_pods.append(
+                dc_pod_factory(
+                    interface=constants.CEPHBLOCKPOOL,
+                    pvc=rbd_pvc,
+                )
+            )
+
+        for cephfs_pvc in cephfs_pvcs:
+            self.dc_pods.append(
+                dc_pod_factory(
+                    interface=constants.CEPHFILESYSTEM,
+                    pvc=cephfs_pvc,
+                )
+            )
+        dd_cmd = f"dd if=/dev/urandom of=/tmp/{self.filename} bs=4M count=3"
+        run_cmd(cmd=dd_cmd)
+        for pod_obj in self.dc_pods:
+            pod.upload(
+                pod_obj.name,
+                f"/tmp/{self.filename}",
+                "/mnt/",
+                namespace=project.namespace,
+            )
+            # Calculate md5sum
+            md5sum = pod.cal_md5sum(pod_obj, self.filename)
+            pod_obj.pvc.md5sum = md5sum
+        logger.info("IO completed on all pods")
+
+        logger.info("Running ios on OBCs")
+        self.sanity_helpers.obc_put_obj_create_delete(mcg_obj, bucket_factory)
```

Comment:
This is a setup() and did not enter the actual test steps yet.  
You are creating and deleting the same created objects on an idle cluster, do we really need this line?

Reply:
ack, Made the change. i had forgot to change it.